### PR TITLE
[Pygen]: minor fixes

### PR DESCRIPTION
**Pull request by hodjat91**
This PR fixes a few issues in riscv_asm_program_gen and riscv_instr_gen_config

### CODE REVIEW

File: pygen/pygen_src/riscv_asm_program_gen.py [hunk 1]
```diff
@@ -116,8 +116,8 @@ def gen_program(self):
             self.gen_program_end(hart)
             for hart in range(cfg.num_of_harts):
                 self.gen_data_page_begin(hart)
-                if(cfg.no_data_page):
-                    self.gen_data_page()
+                if not cfg.no_data_page:
```
Comment (hodjat91) [inline pygen/pygen_src/riscv_asm_program_gen.py:119]: Hi @aneels3 
Please confirm.

File: pygen/pygen_src/riscv_instr_gen_config.py [hunk 2]
```diff
@@ -111,16 +111,16 @@ def __init__(self, argv):
         self.num_of_harts = argv.num_of_harts
         self.fix_sp = argv.fix_sp
         self.use_push_data_section = argv.use_push_data_section
-        self.boot_mode_opts = argv.boot_mode_opts
+        self.boot_mode_opts = argv.boot_mode
```
Comment (hodjat91) [inline pygen/pygen_src/riscv_instr_gen_config.py:114]: Changing this for integration purposes to make it compatible with the overall run flow (`+boot_mode` has been used on SV side). @aneels3

File: pygen/pygen_src/riscv_instr_gen_config.py [hunk 3]
```diff
@@ -111,16 +111,16 @@ def __init__(self, argv):
...
+        if self.boot_mode_opts:
             logging.info("Got boot mode option - %0s", self.boot_mode_opts)
-            if(self.boot_mode_opts == "m"):
-                self.init_privileged_mode = privileged_mode_t.MACHINE_MODE.name
-            elif(self.boot_mode_opts == "s"):
-                self.init_privileged_mode = privileged_mode_t.SUPERVISOR_MODE.name
-            elif(self.boot_mode_opts == "u"):
-                self.init_privileged_mode = privileged_mode_t.USER_MODE.name
+            if self.boot_mode_opts == "m":
+                self.init_privileged_mode = privileged_mode_t.MACHINE_MODE
```
Comment (hodjat91) [inline pygen/pygen_src/riscv_instr_gen_config.py:119]: `self.init_privileged_mode` should be from type `Enum` and not `str`. @aneels3, please confirm.

File: pygen/pygen_src/riscv_asm_program_gen.py (hunk 1 above)
Comment (aneels3) [reply pygen/pygen_src/riscv_asm_program_gen.py:119]: Hi @hodjat91
I think the commented part should be `self.gen_data_page(hart)`

rest LGTM, thanks!

File: pygen/pygen_src/riscv_instr_gen_config.py (hunk 3 above)
Comment (aneels3) [reply pygen/pygen_src/riscv_instr_gen_config.py:119]: This looks good.
We have to change the string comparison in the whole flow. for eg 
`if "USER_MODE" == self.init_privileged_mode:` this won't work.
I am thinking to create a separate PR for this change such that there won't be any conflicts within the generator.

You can definitely move forward with these changes. I will resolve the possible conflicts in the next PR.

File: pygen/pygen_src/riscv_instr_gen_config.py (hunk 2 above)
Comment (aneels3) [reply pygen/pygen_src/riscv_instr_gen_config.py:114]: Okay great!
LGTM!

**Review by aneels3 — APPROVED** [review]

File: pygen/pygen_src/riscv_instr_gen_config.py (hunk 3 above)
Comment (hodjat91) [reply pygen/pygen_src/riscv_instr_gen_config.py:119]: Sounds great, thanks @aneels3

**Pull request merged**